packages/apps/api/src/handlers/user.ts

```diff
@@ -113,42 +91,40 @@ export async function createUser(
     !isNonBlank(email) ||
     !isNonBlank(password)
   ) {
-    response
-      .status(400)
-      .json(toMessage('All fields must be provided and non-blank'));
-    return;
+    throw new HttpError({
+      status: 400,
+      message: 'All fields must be provided and non-blank',
+    });
   }
 
   const sierraGet: APIResponse<PatronRecord> = await sierraClient.getPatronRecordByEmail(
     email
   );
-  if (sierraGet.status !== ResponseStatus.NotFound) {
-    if (sierraGet.status === ResponseStatus.Success) {
-      response
-        .status(409)
-        .json(
-          toMessage('Patron record with email [' + email + '] already exists')
-        );
-    } else {
-      response.status(500).json(toMessage(sierraGet.message));
-    }
-    return;
+  if (sierraGet.status === ResponseStatus.Success) {
+    throw new HttpError({
+      status: 409,
+      message: `Patron record with email [${email}] already exists`,
+    });
+  } else if (sierraGet.status !== ResponseStatus.NotFound) {
+    throw new HttpError({
+      status: 500,
+      message: 'Something went wrong: ' + sierraGet.message,
+    });
```

Comment:
Possibly not for addressing here, but elsewhere we don't include any context in 500 errors – we return a generic error and log the message.